### PR TITLE
Refactor quiz classes

### PR DESCRIPTION
**Pull request by kdeloach**
This merges Question and MultiChoiceQuestion (all questions are
multiple choice). This also adds strongly typed classes for the answer
field to differentiate between checkbox & radio button questions.

This is not a priority for user testing.

### CODE REVIEW

File: src/nyc_trees/apps/home/training/types.py [hunk 1]
```diff
@@ -210,38 +210,40 @@ def score(self, answers):
...
-        super(Question, self).__init__(text, choices, [answer])
+    def __init__(self, *answers):
+        self.answers = list(answers)
+
 
-    def is_multiple_choice(self):
-        return False
+class SingleAnswer(MultipleAnswer):
+    """
+    answer - int
```
Comment (kdeloach) [inline src/nyc_trees/apps/home/training/types.py]: TODO: Documentation

File: src/nyc_trees/apps/home/training/types.py [hunk 2]
```diff
@@ -210,38 +210,40 @@ def score(self, answers):
...
-        self.answer = list(answer)
+        self.answer = answer
 
     def is_correct(self, candidate):
-        return candidate and set(candidate) == set(self.answer)
+        return candidate and set(candidate) == set(self.answer.answers)
 
-    def is_multiple_choice(self):
-        return True
+    def allow_multiple_selections(self):
```
Comment (steventlamb) [inline src/nyc_trees/apps/home/training/types.py]: how about `has_multiple_answers`

Comment (kdeloach) [inline src/nyc_trees/apps/home/training/types.py]: The rationale behind this name is that it's possible to allow multiple selections (checkbox) but only have 1 "answer". I think either name is an improvement though.

**Comment by steventlamb** [comment]
+1 works and makes sense

**Pull request merged**